src/frontend/src/functions/auth.tsx

```diff
@@ -17,6 +17,7 @@ import { useUserState } from '../states/UserState';
 
 export const doClassicLogin = async (username: string, password: string) => {
   const { host } = useLocalState.getState();
+  const apiState = useServerApiState.getState();
```

Comment:
Question: Why is the full state accesed here? The following would be more precise
```suggestion
  const [setAuthenticated] = useServerApiState((state) => [state.setAuthenticated]);
```